cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1364,15 +1364,41 @@ impl fmt::Debug for Inst {
     }
 }
 
+/// Is the given (AluRmiROpcode, src, dst) combination causing a redefinition of the input operand,
+/// instead of an actual use?
+fn alu_rmi_r_is_def(src: &RegMemImm, dst: Writable<Reg>, op: AluRmiROpcode) -> bool {
+    op == AluRmiROpcode::Xor
+        && if let RegMemImm::Reg { reg: src } = src {
```

Comment:
This reads a little awkwardly (three cheers for expression-`if`s, but...). Perhaps add a helper to `RegMemImm` such as `.to_reg() -> Option<Reg>`, then you can do `op == ... && src.to_reg() == Some(dst.to_reg()`?

Reply:
Another thought: why not make these methods of `Inst`? Then we can use this logic with `inst.is_def()` and avoid searching around for `alu_rmi_r_is_def`?

Reply:
I like the expression-`if`, but maybe it could be made clearer by changing the formal parameters thusly:
```
fn alu_rmi_r_is_def(rmi: &RegMemImm, reg: Writable<Reg>, op: AluRmiROpcode)
```

Other comments:
* I would find the name `alu_rmi_r_is_const` more natural; could that be considered?
* also `op == AluRmiROpcode::Sub` behaves similarly, and really we should handle that too, even though we don't expect anybody to actually use it

Reply:
Or maybe `alu_rmi_r_produces_const` ..

Reply:
Thanks for the comments! To sum up:

- created it as a method of Inst
- called `produces_const()`
- and added helper `to_reg` to both RegMem and RegMemImm